src/client/lazy-app/Compress/Modal/index.tsx

```diff
@@ -0,0 +1,148 @@
+import { h, Component } from 'preact';
+import * as style from './style.css';
+import 'add-css:./style.css';
+import { linkRef } from 'shared/prerendered-app/util';
+import {
+  InfoIcon,
+  DiamondStarIcon,
+  ModalErrorIcon,
+} from 'client/lazy-app/icons';
+
+interface Props {}
+
+export interface ModalMessage {
+  type: 'info' | 'error' | 'update';
+  title: string;
+  content: string;
+}
+
+interface State {
+  message: ModalMessage;
+  shown: boolean;
+}
+
+export default class Modal extends Component<Props, State> {
+  state: State = {
+    message: {
+      type: 'info',
+      title: 'default',
+      content: 'default',
+    },
+    shown: false,
+  };
+
+  private modal?: HTMLElement;
+  private returnFocusElement?: HTMLElement | null;
+
+  showModal(message: ModalMessage) {
+    if (this.state.shown) return;
+    if (!this.modal) return;
+
+    // Set element to return focus to after hiding
+    this.returnFocusElement = document.activeElement as HTMLElement;
+
+    this.modal.style.display = '';
+    this.setState({
+      message: message,
+      shown: true,
+    });
+    // Wait for the 'display' reset to take place, then focus
+    setTimeout(() => {
```

Comment:
```suggestion
    requestAnimationFrame(() => {
```

Reply:
But on second thought, it might actually be more idiomatic to use the `ref` prop to focus the button once a DOM element been created for it. Something like

```js
  <button ref={(el) => el?.focus()} /*...*/>
```

Reply:
I'm confused, will that work in this case? I'm not adding/removing the Modal component to the DOM tree on show/hide, just changing some CSS properties to "hide" it from the user.

Reply:
I think the correct Preact way to do this would be in `componentDidUpdate`. But, you get all this stuff for free with `<dialog>`, so we should just use that.